// react-native-animated-wheel-picker/lib/commonjs/components/Picker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _reactNative = require("react-native");

var _react = _interopRequireDefault(require("react"));

var _reactNativeReanimated = _interopRequireWildcard(require("react-native-reanimated"));

var _reactNativeGestureHandler = require("react-native-gesture-handler");

var _reactNativeRedash = require("react-native-redash");

var _maskedView = _interopRequireDefault(require("@react-native-masked-view/masked-view"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Picker = _ref => {
  let {
    itemHeight = 30,
    pickerData,
    visible = 5,
    textStyle,
    maskedComponents,
    contentContainerStyle,
    initialIndex = 0,
    onSelected,
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, _extends({}, props, {
    style: [styles.container]
  }), /*#__PURE__*/_react.default.createElement(PickerItem, {
    itemHeight: itemHeight,
    pickerData: pickerData,
    visible: visible,
    textStyle: textStyle !== null && textStyle !== void 0 ? textStyle : {},
    maskedComponents: maskedComponents,
    contentContainerStyle: contentContainerStyle,
    initialIndex: initialIndex,
    onSelected: onSelected
  }));
};

const duration = 1000;

const PickerItem = _ref2 => {
  let {
    itemHeight,
    pickerData,
    visible,
    textStyle,
    maskedComponents,
    contentContainerStyle,
    initialIndex,
    onSelected
  } = _ref2;
  const translateY = (0, _reactNativeReanimated.useSharedValue)(-itemHeight * initialIndex);
  const snapPoints = new Array(pickerData.length).fill(0).map((_, i) => i * -itemHeight);
  const timingConfig = {
    duration: duration,
    easing: _reactNativeReanimated.Easing.bezier(0.35, 1, 0.35, 1)
  };

  const wrapper = index => {
    onSelected && onSelected(pickerData[index], index);
  };

  const onGestureEvent = (0, _reactNativeReanimated.useAnimatedGestureHandler)({
    onStart: (_event, ctx) => {
      // @ts-ignore
      ctx.y = translateY.value; // triggered at the start of the pan gesture
    },
    onActive: (_ref3, ctx) => {
      let {
        translationY
      } = _ref3;
      // @ts-ignore
      translateY.value = translationY + ctx.y; // triggered on every frame of the pan gesture
    },
    onEnd: _ref4 => {
      let {
        velocityY
      } = _ref4;
      const snapPointY = (0, _reactNativeRedash.snapPoint)(translateY.value, velocityY, snapPoints);
      const index = Math.abs(snapPointY / itemHeight);
      translateY.value = (0, _reactNativeReanimated.withTiming)(snapPointY, timingConfig);
      (0, _reactNativeReanimated.runOnJS)(wrapper)(index); // triggered at the end of the pan gesture
    }
  });
  const animatedStyle = (0, _reactNativeReanimated.useAnimatedStyle)(() => ({
    transform: [{
      translateY: translateY.value
    }]
  }));
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_maskedView.default, {
    androidRenderingMode: "software",
    maskElement: /*#__PURE__*/_react.default.createElement(_reactNative.View, null, /*#__PURE__*/_react.default.createElement(_reactNativeReanimated.default.View, {
      style: [animatedStyle, contentContainerStyle, {
        height: itemHeight * visible,
        paddingTop: itemHeight * visible / 2 - itemHeight / 2
      }]
    }, pickerData.map((item, index) => /*#__PURE__*/_react.default.createElement(Item, {
      key: index,
      translateY: translateY,
      index: index,
      itemHeight: itemHeight,
      visible: visible,
      data: item,
      textStyle: textStyle
    }))))
  }, maskedComponents ? maskedComponents : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_reactNative.View // eslint-disable-next-line react-native/no-inline-styles
  , {
    style: {
      height: itemHeight * Math.trunc(visible / 2),
      backgroundColor: 'grey'
    }
  }), /*#__PURE__*/_react.default.createElement(_reactNative.View // eslint-disable-next-line react-native/no-inline-styles
  , {
    style: {
      height: itemHeight,
      backgroundColor: 'white'
    }
  }), /*#__PURE__*/_react.default.createElement(_reactNative.View // eslint-disable-next-line react-native/no-inline-styles
  , {
    style: {
      height: itemHeight * Math.trunc(visible / 2),
      backgroundColor: 'grey'
    }
  }))), /*#__PURE__*/_react.default.createElement(_reactNativeGestureHandler.GestureHandlerRootView, {
    style: _reactNative.StyleSheet.absoluteFillObject
  }, /*#__PURE__*/_react.default.createElement(_reactNativeGestureHandler.PanGestureHandler, {
    onGestureEvent: onGestureEvent
  }, /*#__PURE__*/_react.default.createElement(_reactNativeReanimated.default.View, {
    // eslint-disable-next-line react-native/no-inline-styles
    style: {
      flex: 1
    }
  }))));
};

const Item = _ref5 => {
  let {
    translateY,
    index,
    itemHeight,
    visible,
    data,
    textStyle
  } = _ref5;
  // const y = useDerivedValue(() => translateY.value, [translateY.value]);
  const y = (0, _reactNativeReanimated.useDerivedValue)(() => (0, _reactNativeReanimated.interpolate)(translateY.value / -itemHeight, [index - visible / 2, index, index + visible / 2], [-1, 0, 1], _reactNativeReanimated.Extrapolate.CLAMP));
  const textAnimation = (0, _reactNativeReanimated.useAnimatedStyle)(() => {
    return {
      opacity: 1 / (1 + Math.abs(y.value)),
      transform: [{
        scale: 1 - 0.1 * Math.abs(y.value)
      }, {
        perspective: 500
      }, {
        rotateX: `${65 * y.value}deg`
      }]
    };
  });
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.textContainer, {
      height: itemHeight
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNativeReanimated.default.Text, {
    style: [textAnimation, textStyle]
  }, data.title));
};

var _default = Picker;
exports.default = _default;

const styles = _reactNative.StyleSheet.create({
  container: {
    overflow: 'hidden',
    justifyContent: 'center',
    flex: 1
  },
  textContainer: {
    alignItems: 'center',
    justifyContent: 'center'
  }
});
