// react-native-agora/lib/commonjs/IAgoraLog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogLevel = exports.LogFilterType = exports.LogConfig = void 0;
require("./extension/IAgoraLogExtension");
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
/**
 * The output log level of the SDK.
 */
let LogLevel;
/**
 * The output log level of the SDK.
 */
exports.LogLevel = LogLevel;
(function (LogLevel) {
  LogLevel[LogLevel["LogLevelNone"] = 0] = "LogLevelNone";
  LogLevel[LogLevel["LogLevelInfo"] = 1] = "LogLevelInfo";
  LogLevel[LogLevel["LogLevelWarn"] = 2] = "LogLevelWarn";
  LogLevel[LogLevel["LogLevelError"] = 4] = "LogLevelError";
  LogLevel[LogLevel["LogLevelFatal"] = 8] = "LogLevelFatal";
  LogLevel[LogLevel["LogLevelApiCall"] = 16] = "LogLevelApiCall";
})(LogLevel || (exports.LogLevel = LogLevel = {}));
let LogFilterType;
/**
 * Configuration of Agora SDK log files.
 */
exports.LogFilterType = LogFilterType;
(function (LogFilterType) {
  LogFilterType[LogFilterType["LogFilterOff"] = 0] = "LogFilterOff";
  LogFilterType[LogFilterType["LogFilterDebug"] = 2063] = "LogFilterDebug";
  LogFilterType[LogFilterType["LogFilterInfo"] = 15] = "LogFilterInfo";
  LogFilterType[LogFilterType["LogFilterWarn"] = 14] = "LogFilterWarn";
  LogFilterType[LogFilterType["LogFilterError"] = 12] = "LogFilterError";
  LogFilterType[LogFilterType["LogFilterCritical"] = 8] = "LogFilterCritical";
  LogFilterType[LogFilterType["LogFilterMask"] = 2063] = "LogFilterMask";
})(LogFilterType || (exports.LogFilterType = LogFilterType = {}));
class LogConfig {
  constructor() {
    _defineProperty(this, "filePath", void 0);
    _defineProperty(this, "fileSizeInKB", void 0);
    _defineProperty(this, "level", void 0);
  }
}
exports.LogConfig = LogConfig;
