// react-native-agora/lib/commonjs/IAgoraSpatialAudio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SpatialAudioZone = exports.RemoteVoicePositionInfo = exports.ILocalSpatialAudioEngine = exports.IBaseSpatialAudioEngine = void 0;
require("./extension/IAgoraSpatialAudioExtension");
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
/**
 * The spatial position of the remote user or the media player.
 */
class RemoteVoicePositionInfo {
  constructor() {
    _defineProperty(this, "position", void 0);
    _defineProperty(this, "forward", void 0);
  }
}

/**
 * Sound insulation area settings.
 */
exports.RemoteVoicePositionInfo = RemoteVoicePositionInfo;
class SpatialAudioZone {
  constructor() {
    _defineProperty(this, "zoneSetId", void 0);
    _defineProperty(this, "position", void 0);
    _defineProperty(this, "forward", void 0);
    _defineProperty(this, "right", void 0);
    _defineProperty(this, "up", void 0);
    _defineProperty(this, "forwardLength", void 0);
    _defineProperty(this, "rightLength", void 0);
    _defineProperty(this, "upLength", void 0);
    _defineProperty(this, "audioAttenuation", void 0);
  }
}

/**
 * This class contains some of the APIs in the ILocalSpatialAudioEngine class.
 * The ILocalSpatialAudioEngine class inherits from IBaseSpatialAudioEngine.
 */
exports.SpatialAudioZone = SpatialAudioZone;
class IBaseSpatialAudioEngine {}

/**
 * This class calculates user positions through the SDK to implement the spatial audio effect.
 * This class inherits from IBaseSpatialAudioEngine . Before calling other APIs in this class, you need to call the initialize method to initialize this class.
 */
exports.IBaseSpatialAudioEngine = IBaseSpatialAudioEngine;
class ILocalSpatialAudioEngine extends IBaseSpatialAudioEngine {}
exports.ILocalSpatialAudioEngine = ILocalSpatialAudioEngine;
