// react-native-agora/lib/commonjs/IAgoraRhythmPlayer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RhythmPlayerStateType = exports.RhythmPlayerErrorType = exports.AgoraRhythmPlayerConfig = void 0;
require("./extension/IAgoraRhythmPlayerExtension");
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
/**
 * Virtual metronome state.
 */
let RhythmPlayerStateType;
/**
 * Virtual Metronome error message.
 */
exports.RhythmPlayerStateType = RhythmPlayerStateType;
(function (RhythmPlayerStateType) {
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateIdle"] = 810] = "RhythmPlayerStateIdle";
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateOpening"] = 811] = "RhythmPlayerStateOpening";
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateDecoding"] = 812] = "RhythmPlayerStateDecoding";
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStatePlaying"] = 813] = "RhythmPlayerStatePlaying";
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateFailed"] = 814] = "RhythmPlayerStateFailed";
})(RhythmPlayerStateType || (exports.RhythmPlayerStateType = RhythmPlayerStateType = {}));
let RhythmPlayerErrorType;
/**
 * The metronome configuration.
 */
exports.RhythmPlayerErrorType = RhythmPlayerErrorType;
(function (RhythmPlayerErrorType) {
  RhythmPlayerErrorType[RhythmPlayerErrorType["RhythmPlayerErrorOk"] = 0] = "RhythmPlayerErrorOk";
  RhythmPlayerErrorType[RhythmPlayerErrorType["RhythmPlayerErrorFailed"] = 1] = "RhythmPlayerErrorFailed";
  RhythmPlayerErrorType[RhythmPlayerErrorType["RhythmPlayerErrorCanNotOpen"] = 801] = "RhythmPlayerErrorCanNotOpen";
  RhythmPlayerErrorType[RhythmPlayerErrorType["RhythmPlayerErrorCanNotPlay"] = 802] = "RhythmPlayerErrorCanNotPlay";
  RhythmPlayerErrorType[RhythmPlayerErrorType["RhythmPlayerErrorFileOverDurationLimit"] = 803] = "RhythmPlayerErrorFileOverDurationLimit";
})(RhythmPlayerErrorType || (exports.RhythmPlayerErrorType = RhythmPlayerErrorType = {}));
class AgoraRhythmPlayerConfig {
  constructor() {
    _defineProperty(this, "beatsPerMeasure", void 0);
    _defineProperty(this, "beatsPerMinute", void 0);
  }
}
exports.AgoraRhythmPlayerConfig = AgoraRhythmPlayerConfig;
