// react-native-agora/lib/commonjs/IAgoraMusicContentCenter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PreloadStatusCode = exports.MvProperty = exports.MusicContentCenterStatusCode = exports.MusicContentCenterConfiguration = exports.MusicCollection = exports.MusicChartInfo = exports.MusicChartCollection = exports.Music = exports.IMusicPlayer = exports.IMusicContentCenter = exports.ClimaxSegment = void 0;
require("./extension/IAgoraMusicContentCenterExtension");
var _IAgoraMediaPlayer = require("./IAgoraMediaPlayer");
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
/**
 * @ignore
 */
let PreloadStatusCode;
/**
 * @ignore
 */
exports.PreloadStatusCode = PreloadStatusCode;
(function (PreloadStatusCode) {
  PreloadStatusCode[PreloadStatusCode["KPreloadStatusCompleted"] = 0] = "KPreloadStatusCompleted";
  PreloadStatusCode[PreloadStatusCode["KPreloadStatusFailed"] = 1] = "KPreloadStatusFailed";
  PreloadStatusCode[PreloadStatusCode["KPreloadStatusPreloading"] = 2] = "KPreloadStatusPreloading";
})(PreloadStatusCode || (exports.PreloadStatusCode = PreloadStatusCode = {}));
let MusicContentCenterStatusCode;
/**
 * @ignore
 */
exports.MusicContentCenterStatusCode = MusicContentCenterStatusCode;
(function (MusicContentCenterStatusCode) {
  MusicContentCenterStatusCode[MusicContentCenterStatusCode["KMusicContentCenterStatusOk"] = 0] = "KMusicContentCenterStatusOk";
  MusicContentCenterStatusCode[MusicContentCenterStatusCode["KMusicContentCenterStatusErr"] = 1] = "KMusicContentCenterStatusErr";
})(MusicContentCenterStatusCode || (exports.MusicContentCenterStatusCode = MusicContentCenterStatusCode = {}));
class MusicChartInfo {
  constructor() {
    _defineProperty(this, "chartName", void 0);
    _defineProperty(this, "id", void 0);
  }
}

/**
 * @ignore
 */
exports.MusicChartInfo = MusicChartInfo;
class MusicChartCollection {}

/**
 * @ignore
 */
exports.MusicChartCollection = MusicChartCollection;
class MvProperty {
  constructor() {
    _defineProperty(this, "resolution", void 0);
    _defineProperty(this, "bandwidth", void 0);
  }
}

/**
 * The climax parts of the music.
 */
exports.MvProperty = MvProperty;
class ClimaxSegment {
  constructor() {
    _defineProperty(this, "startTimeMs", void 0);
    _defineProperty(this, "endTimeMs", void 0);
  }
}

/**
 * @ignore
 */
exports.ClimaxSegment = ClimaxSegment;
class Music {
  constructor() {
    _defineProperty(this, "songCode", void 0);
    _defineProperty(this, "name", void 0);
    _defineProperty(this, "singer", void 0);
    _defineProperty(this, "poster", void 0);
    _defineProperty(this, "releaseTime", void 0);
    _defineProperty(this, "durationS", void 0);
    _defineProperty(this, "type", void 0);
    _defineProperty(this, "pitchType", void 0);
    _defineProperty(this, "lyricCount", void 0);
    _defineProperty(this, "lyricList", void 0);
    _defineProperty(this, "climaxSegmentCount", void 0);
    _defineProperty(this, "climaxSegmentList", void 0);
    _defineProperty(this, "mvPropertyCount", void 0);
    _defineProperty(this, "mvPropertyList", void 0);
  }
}

/**
 * @ignore
 */
exports.Music = Music;
class MusicCollection {}

/**
 * @ignore
 */
exports.MusicCollection = MusicCollection;
/**
 * @ignore
 */
class MusicContentCenterConfiguration {
  constructor() {
    _defineProperty(this, "appId", void 0);
    _defineProperty(this, "token", void 0);
    _defineProperty(this, "mccUid", void 0);
  }
}

/**
 * @ignore
 */
exports.MusicContentCenterConfiguration = MusicContentCenterConfiguration;
class IMusicPlayer extends _IAgoraMediaPlayer.IMediaPlayer {}

/**
 * @ignore
 */
exports.IMusicPlayer = IMusicPlayer;
class IMusicContentCenter {}
exports.IMusicContentCenter = IMusicContentCenter;
