// react-native-agora/lib/module/AgoraMediaPlayerTypes.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
import './extension/AgoraMediaPlayerTypesExtension';
/**
 * The playback state.
 */
export let MediaPlayerState;

/**
 * Error codes of the media player.
 */
(function (MediaPlayerState) {
  MediaPlayerState[MediaPlayerState["PlayerStateIdle"] = 0] = "PlayerStateIdle";
  MediaPlayerState[MediaPlayerState["PlayerStateOpening"] = 1] = "PlayerStateOpening";
  MediaPlayerState[MediaPlayerState["PlayerStateOpenCompleted"] = 2] = "PlayerStateOpenCompleted";
  MediaPlayerState[MediaPlayerState["PlayerStatePlaying"] = 3] = "PlayerStatePlaying";
  MediaPlayerState[MediaPlayerState["PlayerStatePaused"] = 4] = "PlayerStatePaused";
  MediaPlayerState[MediaPlayerState["PlayerStatePlaybackCompleted"] = 5] = "PlayerStatePlaybackCompleted";
  MediaPlayerState[MediaPlayerState["PlayerStatePlaybackAllLoopsCompleted"] = 6] = "PlayerStatePlaybackAllLoopsCompleted";
  MediaPlayerState[MediaPlayerState["PlayerStateStopped"] = 7] = "PlayerStateStopped";
  MediaPlayerState[MediaPlayerState["PlayerStatePausingInternal"] = 50] = "PlayerStatePausingInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateStoppingInternal"] = 51] = "PlayerStateStoppingInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateSeekingInternal"] = 52] = "PlayerStateSeekingInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateGettingInternal"] = 53] = "PlayerStateGettingInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateNoneInternal"] = 54] = "PlayerStateNoneInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateDoNothingInternal"] = 55] = "PlayerStateDoNothingInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateSetTrackInternal"] = 56] = "PlayerStateSetTrackInternal";
  MediaPlayerState[MediaPlayerState["PlayerStateFailed"] = 100] = "PlayerStateFailed";
})(MediaPlayerState || (MediaPlayerState = {}));
export let MediaPlayerError;

/**
 * The type of the media stream.
 */
(function (MediaPlayerError) {
  MediaPlayerError[MediaPlayerError["PlayerErrorNone"] = 0] = "PlayerErrorNone";
  MediaPlayerError[MediaPlayerError["PlayerErrorInvalidArguments"] = -1] = "PlayerErrorInvalidArguments";
  MediaPlayerError[MediaPlayerError["PlayerErrorInternal"] = -2] = "PlayerErrorInternal";
  MediaPlayerError[MediaPlayerError["PlayerErrorNoResource"] = -3] = "PlayerErrorNoResource";
  MediaPlayerError[MediaPlayerError["PlayerErrorInvalidMediaSource"] = -4] = "PlayerErrorInvalidMediaSource";
  MediaPlayerError[MediaPlayerError["PlayerErrorUnknownStreamType"] = -5] = "PlayerErrorUnknownStreamType";
  MediaPlayerError[MediaPlayerError["PlayerErrorObjNotInitialized"] = -6] = "PlayerErrorObjNotInitialized";
  MediaPlayerError[MediaPlayerError["PlayerErrorCodecNotSupported"] = -7] = "PlayerErrorCodecNotSupported";
  MediaPlayerError[MediaPlayerError["PlayerErrorVideoRenderFailed"] = -8] = "PlayerErrorVideoRenderFailed";
  MediaPlayerError[MediaPlayerError["PlayerErrorInvalidState"] = -9] = "PlayerErrorInvalidState";
  MediaPlayerError[MediaPlayerError["PlayerErrorUrlNotFound"] = -10] = "PlayerErrorUrlNotFound";
  MediaPlayerError[MediaPlayerError["PlayerErrorInvalidConnectionState"] = -11] = "PlayerErrorInvalidConnectionState";
  MediaPlayerError[MediaPlayerError["PlayerErrorSrcBufferUnderflow"] = -12] = "PlayerErrorSrcBufferUnderflow";
  MediaPlayerError[MediaPlayerError["PlayerErrorInterrupted"] = -13] = "PlayerErrorInterrupted";
  MediaPlayerError[MediaPlayerError["PlayerErrorNotSupported"] = -14] = "PlayerErrorNotSupported";
  MediaPlayerError[MediaPlayerError["PlayerErrorTokenExpired"] = -15] = "PlayerErrorTokenExpired";
  MediaPlayerError[MediaPlayerError["PlayerErrorIpExpired"] = -16] = "PlayerErrorIpExpired";
  MediaPlayerError[MediaPlayerError["PlayerErrorUnknown"] = -17] = "PlayerErrorUnknown";
})(MediaPlayerError || (MediaPlayerError = {}));
export let MediaStreamType;

/**
 * Media player events.
 */
(function (MediaStreamType) {
  MediaStreamType[MediaStreamType["StreamTypeUnknown"] = 0] = "StreamTypeUnknown";
  MediaStreamType[MediaStreamType["StreamTypeVideo"] = 1] = "StreamTypeVideo";
  MediaStreamType[MediaStreamType["StreamTypeAudio"] = 2] = "StreamTypeAudio";
  MediaStreamType[MediaStreamType["StreamTypeSubtitle"] = 3] = "StreamTypeSubtitle";
})(MediaStreamType || (MediaStreamType = {}));
export let MediaPlayerEvent;

/**
 * Events that occur when media resources are preloaded.
 */
(function (MediaPlayerEvent) {
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSeekBegin"] = 0] = "PlayerEventSeekBegin";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSeekComplete"] = 1] = "PlayerEventSeekComplete";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSeekError"] = 2] = "PlayerEventSeekError";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventAudioTrackChanged"] = 5] = "PlayerEventAudioTrackChanged";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventBufferLow"] = 6] = "PlayerEventBufferLow";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventBufferRecover"] = 7] = "PlayerEventBufferRecover";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventFreezeStart"] = 8] = "PlayerEventFreezeStart";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventFreezeStop"] = 9] = "PlayerEventFreezeStop";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSwitchBegin"] = 10] = "PlayerEventSwitchBegin";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSwitchComplete"] = 11] = "PlayerEventSwitchComplete";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSwitchError"] = 12] = "PlayerEventSwitchError";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventFirstDisplayed"] = 13] = "PlayerEventFirstDisplayed";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventReachCacheFileMaxCount"] = 14] = "PlayerEventReachCacheFileMaxCount";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventReachCacheFileMaxSize"] = 15] = "PlayerEventReachCacheFileMaxSize";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventTryOpenStart"] = 16] = "PlayerEventTryOpenStart";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventTryOpenSucceed"] = 17] = "PlayerEventTryOpenSucceed";
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventTryOpenFailed"] = 18] = "PlayerEventTryOpenFailed";
})(MediaPlayerEvent || (MediaPlayerEvent = {}));
export let PlayerPreloadEvent;

/**
 * The detailed information of the media stream.
 */
(function (PlayerPreloadEvent) {
  PlayerPreloadEvent[PlayerPreloadEvent["PlayerPreloadEventBegin"] = 0] = "PlayerPreloadEventBegin";
  PlayerPreloadEvent[PlayerPreloadEvent["PlayerPreloadEventComplete"] = 1] = "PlayerPreloadEventComplete";
  PlayerPreloadEvent[PlayerPreloadEvent["PlayerPreloadEventError"] = 2] = "PlayerPreloadEventError";
})(PlayerPreloadEvent || (PlayerPreloadEvent = {}));
export class PlayerStreamInfo {
  constructor() {
    _defineProperty(this, "streamIndex", void 0);
    _defineProperty(this, "streamType", void 0);
    _defineProperty(this, "codecName", void 0);
    _defineProperty(this, "language", void 0);
    _defineProperty(this, "videoFrameRate", void 0);
    _defineProperty(this, "videoBitRate", void 0);
    _defineProperty(this, "videoWidth", void 0);
    _defineProperty(this, "videoHeight", void 0);
    _defineProperty(this, "videoRotation", void 0);
    _defineProperty(this, "audioSampleRate", void 0);
    _defineProperty(this, "audioChannels", void 0);
    _defineProperty(this, "audioBitsPerSample", void 0);
    _defineProperty(this, "duration", void 0);
  }
}

/**
 * Information about the video bitrate of the media resource being played.
 */
export class SrcInfo {
  constructor() {
    _defineProperty(this, "bitrateInKbps", void 0);
    _defineProperty(this, "name", void 0);
  }
}

/**
 * The type of media metadata.
 */
export let MediaPlayerMetadataType;

/**
 * Statistics about the media files being cached.
 */
(function (MediaPlayerMetadataType) {
  MediaPlayerMetadataType[MediaPlayerMetadataType["PlayerMetadataTypeUnknown"] = 0] = "PlayerMetadataTypeUnknown";
  MediaPlayerMetadataType[MediaPlayerMetadataType["PlayerMetadataTypeSei"] = 1] = "PlayerMetadataTypeSei";
})(MediaPlayerMetadataType || (MediaPlayerMetadataType = {}));
export class CacheStatistics {
  constructor() {
    _defineProperty(this, "fileSize", void 0);
    _defineProperty(this, "cacheSize", void 0);
    _defineProperty(this, "downloadSize", void 0);
  }
}

/**
 * Information related to the media player.
 */
export class PlayerUpdatedInfo {
  constructor() {
    _defineProperty(this, "playerId", void 0);
    _defineProperty(this, "deviceId", void 0);
    _defineProperty(this, "cacheStatistics", void 0);
  }
}

/**
 * Information related to the media file to be played and the playback scenario configurations.
 */
export class MediaSource {
  constructor() {
    _defineProperty(this, "url", void 0);
    _defineProperty(this, "uri", void 0);
    _defineProperty(this, "startPos", void 0);
    _defineProperty(this, "autoPlay", void 0);
    _defineProperty(this, "enableCache", void 0);
    _defineProperty(this, "isAgoraSource", void 0);
    _defineProperty(this, "isLiveSource", void 0);
  }
}
