// react-native-agora/lib/module/IAgoraLog.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
import './extension/IAgoraLogExtension';
/**
 * The output log level of the SDK.
 */
export let LogLevel;

/**
 * The output log level of the SDK.
 */
(function (LogLevel) {
  LogLevel[LogLevel["LogLevelNone"] = 0] = "LogLevelNone";
  LogLevel[LogLevel["LogLevelInfo"] = 1] = "LogLevelInfo";
  LogLevel[LogLevel["LogLevelWarn"] = 2] = "LogLevelWarn";
  LogLevel[LogLevel["LogLevelError"] = 4] = "LogLevelError";
  LogLevel[LogLevel["LogLevelFatal"] = 8] = "LogLevelFatal";
  LogLevel[LogLevel["LogLevelApiCall"] = 16] = "LogLevelApiCall";
})(LogLevel || (LogLevel = {}));
export let LogFilterType;

/**
 * Configuration of Agora SDK log files.
 */
(function (LogFilterType) {
  LogFilterType[LogFilterType["LogFilterOff"] = 0] = "LogFilterOff";
  LogFilterType[LogFilterType["LogFilterDebug"] = 2063] = "LogFilterDebug";
  LogFilterType[LogFilterType["LogFilterInfo"] = 15] = "LogFilterInfo";
  LogFilterType[LogFilterType["LogFilterWarn"] = 14] = "LogFilterWarn";
  LogFilterType[LogFilterType["LogFilterError"] = 12] = "LogFilterError";
  LogFilterType[LogFilterType["LogFilterCritical"] = 8] = "LogFilterCritical";
  LogFilterType[LogFilterType["LogFilterMask"] = 2063] = "LogFilterMask";
})(LogFilterType || (LogFilterType = {}));
export class LogConfig {
  constructor() {
    _defineProperty(this, "filePath", void 0);
    _defineProperty(this, "fileSizeInKB", void 0);
    _defineProperty(this, "level", void 0);
  }
}
