// react-native-ad-manager/lib/commonjs/CTKAdManagerBanner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Banner = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _utils = require("./utils");
var _Constants = require("./Constants");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const ComponentName = 'CTKBannerView';
const AdManagerBannerView = _reactNative.UIManager.getViewManagerConfig(ComponentName) != null ? (0, _reactNative.requireNativeComponent)(ComponentName) : () => {
  throw new Error(_Constants.LINKING_ERROR);
};
class Banner extends _react.default.Component {
  constructor(props) {
    super(props);
    this.handleSizeChange = this.handleSizeChange.bind(this);
    this.state = {
      style: {}
    };
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (Object.entries(this.state.style).toString() === Object.entries(nextState.style).toString() && Object.entries(this.props).toString() === Object.entries(nextProps).toString()) {
      return false;
    }
    return true;
  }
  componentDidMount() {
    this.loadBanner();
  }
  loadBanner() {
    _reactNative.UIManager.dispatchViewManagerCommand((0, _reactNative.findNodeHandle)(this), _reactNative.UIManager.getViewManagerConfig(ComponentName).Commands.loadBanner, []);
  }
  handleSizeChange(_ref) {
    let {
      nativeEvent
    } = _ref;
    const {
      height,
      width
    } = nativeEvent;
    this.setState({
      style: {
        width,
        height
      }
    });
    this.props.onSizeChange && this.props.onSizeChange(nativeEvent);
  }
  render() {
    return /*#__PURE__*/_react.default.createElement(AdManagerBannerView, _extends({}, this.props, {
      style: [this.props.style, this.state.style],
      onSizeChange: this.handleSizeChange,
      onAdLoaded: event => this.props.onAdLoaded && this.props.onAdLoaded(event.nativeEvent),
      onAdFailedToLoad: event => this.props.onAdFailedToLoad && this.props.onAdFailedToLoad((0, _utils.createErrorFromErrorData)(event.nativeEvent.error)),
      onAppEvent: event => this.props.onAppEvent && this.props.onAppEvent(event.nativeEvent),
      onAdOpened: event => this.props.onAdOpened && this.props.onAdOpened(event.nativeEvent),
      onAdClosed: event => this.props.onAdClosed && this.props.onAdClosed(event.nativeEvent)
    }));
  }
}
exports.Banner = Banner;
