// react-native-ad-manager/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Interstitial: true,
  Banner: true,
  AdaptiveBanner: true,
  NativeAdsManager: true,
  TriggerableView: true,
  withNativeAd: true
};
Object.defineProperty(exports, "AdaptiveBanner", {
  enumerable: true,
  get: function () {
    return _CTKAdManagerAdaptiveBanner.AdaptiveBanner;
  }
});
Object.defineProperty(exports, "Banner", {
  enumerable: true,
  get: function () {
    return _CTKAdManagerBanner.Banner;
  }
});
Object.defineProperty(exports, "Interstitial", {
  enumerable: true,
  get: function () {
    return _CTKAdManagerInterstitial.default;
  }
});
Object.defineProperty(exports, "NativeAdsManager", {
  enumerable: true,
  get: function () {
    return _NativeAdsManager.NativeAdsManager;
  }
});
Object.defineProperty(exports, "TriggerableView", {
  enumerable: true,
  get: function () {
    return _TriggerableViewManager.TriggerableView;
  }
});
Object.defineProperty(exports, "withNativeAd", {
  enumerable: true,
  get: function () {
    return _withNativeAd.default;
  }
});
var _CTKAdManagerInterstitial = _interopRequireDefault(require("./CTKAdManagerInterstitial"));
var _CTKAdManagerBanner = require("./CTKAdManagerBanner");
var _CTKAdManagerAdaptiveBanner = require("./CTKAdManagerAdaptiveBanner");
var _NativeAdsManager = require("./native-ads/NativeAdsManager");
var _TriggerableViewManager = require("./native-ads/TriggerableViewManager");
var _withNativeAd = _interopRequireDefault(require("./native-ads/withNativeAd"));
var _AdManagerEvent = require("./AdManagerEvent");
Object.keys(_AdManagerEvent).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _AdManagerEvent[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _AdManagerEvent[key];
    }
  });
});
var _AdManagerTypes = require("./AdManagerTypes");
Object.keys(_AdManagerTypes).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _AdManagerTypes[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _AdManagerTypes[key];
    }
  });
});
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
