// react-native-agora/lib/commonjs/internal/IrisApiEngine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EVENT_TYPE = exports.EVENT_PROCESSORS = exports.DeviceEventEmitter = void 0;
exports.callIrisApi = callIrisApi;
exports.emitEvent = emitEvent;
exports.isDebuggable = isDebuggable;
exports.setDebuggable = setDebuggable;
var _reactNative = require("react-native");
var _base64Js = _interopRequireDefault(require("base64-js"));
var _buffer = require("buffer");
var _AgoraBaseImpl = require("../impl/AgoraBaseImpl");
var _AgoraMediaBaseImpl = require("../impl/AgoraMediaBaseImpl");
var _IAgoraRtcEngineImpl = require("../impl/IAgoraRtcEngineImpl");
var _IAgoraMediaPlayerImpl = require("../impl/IAgoraMediaPlayerImpl");
var _IAgoraMediaPlayerSourceImpl = require("../impl/IAgoraMediaPlayerSourceImpl");
var _IAgoraMusicContentCenterImpl = require("../impl/IAgoraMusicContentCenterImpl");
var _MediaEngineInternal = require("./MediaEngineInternal");
var _MediaPlayerInternal = require("./MediaPlayerInternal");
var _MediaRecorderInternal = require("./MediaRecorderInternal");
var _MusicContentCenterInternal = require("./MusicContentCenterInternal");
var _RtcEngineExInternal = require("./RtcEngineExInternal");
var _EventEmitter = _interopRequireDefault(require("./emitter/EventEmitter"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const JSON = require('json-bigint');
// @ts-ignore
const DeviceEventEmitter = new _EventEmitter.default();
exports.DeviceEventEmitter = DeviceEventEmitter;
const {
  AgoraRtcNg
} = _reactNative.NativeModules;
const AgoraEventEmitter = new _reactNative.NativeEventEmitter(AgoraRtcNg);
AgoraEventEmitter.addListener('AgoraRtcNg:onEvent', handleEvent);
let debuggable = false;

/**
 * @internal
 */
function setDebuggable(flag) {
  debuggable = flag;
}

/**
 * @internal
 */
function isDebuggable() {
  return debuggable && __DEV__;
}

/**
 * @internal
 */
let EVENT_TYPE;
/**
 * @internal
 */
exports.EVENT_TYPE = EVENT_TYPE;
(function (EVENT_TYPE) {
  EVENT_TYPE[EVENT_TYPE["IMediaEngine"] = 0] = "IMediaEngine";
  EVENT_TYPE[EVENT_TYPE["IMediaPlayer"] = 1] = "IMediaPlayer";
  EVENT_TYPE[EVENT_TYPE["IMediaRecorder"] = 2] = "IMediaRecorder";
  EVENT_TYPE[EVENT_TYPE["IRtcEngine"] = 3] = "IRtcEngine";
  EVENT_TYPE[EVENT_TYPE["IMusicContentCenter"] = 4] = "IMusicContentCenter";
})(EVENT_TYPE || (exports.EVENT_TYPE = EVENT_TYPE = {}));
const EVENT_PROCESSORS = {
  IAudioFrameObserver: {
    suffix: 'AudioFrameObserver_',
    type: EVENT_TYPE.IMediaEngine,
    func: [_AgoraMediaBaseImpl.processIAudioFrameObserver, _AgoraMediaBaseImpl.processIAudioFrameObserverBase],
    preprocess: (event, data, buffers) => {
      if (data.audioFrame) {
        data.audioFrame.buffer = buffers[0];
      }
    },
    handlers: () => _MediaEngineInternal.MediaEngineInternal._audio_frame_observers
  },
  IVideoFrameObserver: {
    suffix: 'VideoFrameObserver_',
    type: EVENT_TYPE.IMediaEngine,
    func: [_AgoraMediaBaseImpl.processIVideoFrameObserver],
    preprocess: (event, data, buffers) => {
      if (data.videoFrame) {
        data.videoFrame.yBuffer = buffers[0];
        data.videoFrame.uBuffer = buffers[1];
        data.videoFrame.vBuffer = buffers[2];
        data.videoFrame.metadata_buffer = buffers[3];
        data.videoFrame.alphaBuffer = buffers[4];
      }
    },
    handlers: () => _MediaEngineInternal.MediaEngineInternal._video_frame_observers
  },
  IAudioSpectrumObserver: {
    suffix: 'AudioSpectrumObserver_',
    type: EVENT_TYPE.IRtcEngine,
    func: [_AgoraMediaBaseImpl.processIAudioSpectrumObserver],
    preprocess: (event, data, buffers) => {
      // if (data.data) {
      //   (data.data as AudioSpectrumData).audioSpectrumData = buffers[0];
      // }
    },
    handlers: data => data.playerId === 0 ? _RtcEngineExInternal.RtcEngineExInternal._audio_spectrum_observers : undefined
  },
  IMediaPlayerAudioSpectrumObserver: {
    suffix: 'AudioSpectrumObserver_',
    type: EVENT_TYPE.IMediaPlayer,
    func: [_AgoraMediaBaseImpl.processIAudioSpectrumObserver],
    preprocess: (event, data, buffers) => {
      // if (data.data) {
      //   (data.data as AudioSpectrumData).audioSpectrumData = buffers[0];
      // }
    },
    handlers: data => data.playerId !== 0 ? _MediaPlayerInternal.MediaPlayerInternal._audio_spectrum_observers.get(data.playerId) : undefined
  },
  IAudioEncodedFrameObserver: {
    suffix: 'AudioEncodedFrameObserver_',
    type: EVENT_TYPE.IRtcEngine,
    func: [_AgoraBaseImpl.processIAudioEncodedFrameObserver],
    preprocess: (event, data, buffers) => {
      switch (event) {
        case 'OnRecordAudioEncodedFrame':
        case 'OnPlaybackAudioEncodedFrame':
        case 'OnMixedAudioEncodedFrame':
          data.frameBuffer = buffers[0];
          break;
      }
    },
    handlers: () => _RtcEngineExInternal.RtcEngineExInternal._audio_encoded_frame_observers
  },
  IVideoEncodedFrameObserver: {
    suffix: 'VideoEncodedFrameObserver_',
    type: EVENT_TYPE.IMediaEngine,
    func: [_AgoraMediaBaseImpl.processIVideoEncodedFrameObserver],
    preprocess: (event, data, buffers) => {
      switch (event) {
        case 'onEncodedVideoFrameReceived':
          data.imageBuffer = buffers[0];
          break;
      }
    },
    handlers: () => _MediaEngineInternal.MediaEngineInternal._video_encoded_frame_observers
  },
  IMediaPlayerSourceObserver: {
    suffix: 'MediaPlayerSourceObserver_',
    type: EVENT_TYPE.IMediaPlayer,
    func: [_IAgoraMediaPlayerSourceImpl.processIMediaPlayerSourceObserver],
    handlers: data => _MediaPlayerInternal.MediaPlayerInternal._source_observers.get(data.playerId)
  },
  IMediaPlayerAudioFrameObserver: {
    suffix: 'MediaPlayer_AudioFrameObserver_',
    type: EVENT_TYPE.IMediaPlayer,
    func: [_IAgoraMediaPlayerImpl.processIMediaPlayerAudioFrameObserver],
    preprocess: (event, data, buffers) => {
      if (data.frame) {
        data.frame.data_ = Array.from(buffers[0] ?? []);
      }
    },
    handlers: data => _MediaPlayerInternal.MediaPlayerInternal._audio_frame_observers.get(data.playerId)
  },
  IMediaPlayerVideoFrameObserver: {
    suffix: 'MediaPlayer_VideoFrameObserver_',
    type: EVENT_TYPE.IMediaPlayer,
    func: [_IAgoraMediaPlayerImpl.processIMediaPlayerVideoFrameObserver],
    preprocess: (event, data, buffers) => {
      if (data.frame) {
        data.frame.yBuffer = buffers[0];
        data.frame.uBuffer = buffers[1];
        data.frame.vBuffer = buffers[2];
        data.frame.metadata_buffer = buffers[3];
        data.frame.alphaBuffer = buffers[4];
      }
    },
    handlers: data => _MediaPlayerInternal.MediaPlayerInternal._video_frame_observers.get(data.playerId)
  },
  IMediaRecorderObserver: {
    suffix: 'MediaRecorderObserver_',
    type: EVENT_TYPE.IMediaRecorder,
    func: [_AgoraMediaBaseImpl.processIMediaRecorderObserver],
    handlers: data => [_MediaRecorderInternal.MediaRecorderInternal._observers.get((data.connection.channelId ?? '') + data.connection.localUid)]
  },
  IMetadataObserver: {
    suffix: 'MetadataObserver_',
    type: EVENT_TYPE.IRtcEngine,
    func: [_IAgoraRtcEngineImpl.processIMetadataObserver],
    preprocess: (event, data, buffers) => {
      switch (event) {
        case 'onMetadataReceived':
          if (data.metadata) {
            data.metadata.buffer = buffers[0];
          }
          break;
      }
    },
    handlers: () => _RtcEngineExInternal.RtcEngineExInternal._metadata_observer
  },
  IDirectCdnStreamingEventHandler: {
    suffix: 'DirectCdnStreamingEventHandler_',
    type: EVENT_TYPE.IRtcEngine,
    func: [_IAgoraRtcEngineImpl.processIDirectCdnStreamingEventHandler],
    handlers: () => _RtcEngineExInternal.RtcEngineExInternal._direct_cdn_streaming_event_handler
  },
  IRtcEngineEventHandler: {
    suffix: 'RtcEngineEventHandler_',
    type: EVENT_TYPE.IRtcEngine,
    func: [_IAgoraRtcEngineImpl.processIRtcEngineEventHandler],
    preprocess: (event, data, buffers) => {
      switch (event) {
        case 'onStreamMessage':
        case 'onStreamMessageEx':
          data.data = buffers[0];
          break;
      }
    },
    handlers: () => _RtcEngineExInternal.RtcEngineExInternal._event_handlers
  },
  IMusicContentCenterEventHandler: {
    suffix: 'MusicContentCenterEventHandler_',
    type: EVENT_TYPE.IMusicContentCenter,
    func: [_IAgoraMusicContentCenterImpl.processIMusicContentCenterEventHandler],
    preprocess: (event, data, buffers) => {
      switch (event) {
        case 'onMusicCollectionResult':
          {
            const result = data.result;
            data.result = new _MusicContentCenterInternal.MusicCollectionInternal(result);
            break;
          }
      }
    },
    handlers: () => _MusicContentCenterInternal.MusicContentCenterInternal._event_handlers
  }
};
exports.EVENT_PROCESSORS = EVENT_PROCESSORS;
function handleEvent(_ref) {
  var _processor$handlers;
  let {
    event,
    data,
    buffers
  } = _ref;
  if (debuggable) {
    console.info('onEvent', event, data, buffers);
  }
  let _data;
  try {
    _data = JSON.parse(data) ?? {};
  } catch (e) {
    _data = {};
  }
  let _event = event;
  let processor = EVENT_PROCESSORS.IRtcEngineEventHandler;
  Object.values(EVENT_PROCESSORS).some(it => {
    // @ts-ignore
    const p = it;
    if (_event.startsWith(p.suffix) && processor.handlers(_data) !== undefined) {
      processor = p;
      const reg = new RegExp(`^${processor.suffix}`, 'g');
      _event = _event.replace(reg, '');
      return true;
    }
    return false;
  });
  if (_event.endsWith('Ex')) {
    _event = _event.replace(/Ex$/g, '');
  }
  const _buffers = buffers === null || buffers === void 0 ? void 0 : buffers.map(value => {
    return _buffer.Buffer.from(value, 'base64');
  });
  if (processor.preprocess) processor.preprocess(_event, _data, _buffers);
  (_processor$handlers = processor.handlers(_data)) === null || _processor$handlers === void 0 ? void 0 : _processor$handlers.map(value => {
    if (value) {
      processor.func.map(it => {
        it(value, _event, _data);
      });
    }
  });
  emitEvent(_event, processor.type, _data);
}

/**
 * @internal
 */
function callIrisApi(funcName, params) {
  try {
    const buffers = [];
    if (funcName.startsWith('MediaEngine_')) {
      switch (funcName) {
        case 'MediaEngine_pushAudioFrame':
        case 'MediaEngine_pushCaptureAudioFrame':
        case 'MediaEngine_pushReverseAudioFrame':
        case 'MediaEngine_pushDirectAudioFrame':
          // frame.buffer
          buffers.push(_base64Js.default.fromByteArray(params.frame.buffer));
          break;
        case 'MediaEngine_pushVideoFrame':
          // frame.buffer
          buffers.push(_base64Js.default.fromByteArray(params.frame.buffer));
          // frame.eglContext
          buffers.push(_base64Js.default.fromByteArray(_buffer.Buffer.from('')));
          // frame.metadata_buffer
          buffers.push(_base64Js.default.fromByteArray(_buffer.Buffer.from('')));
          break;
        case 'MediaEngine_pushEncodedVideoImage':
          // imageBuffer
          buffers.push(_base64Js.default.fromByteArray(params.imageBuffer));
          break;
      }
    } else if (funcName.startsWith('MediaPlayer_') || funcName.startsWith('MusicPlayer_')) {
      var _params$toJSON;
      // @ts-ignore
      params.mediaPlayerId = this.getMediaPlayerId();
      const json = (_params$toJSON = params.toJSON) === null || _params$toJSON === void 0 ? void 0 : _params$toJSON.call();
      params.toJSON = function () {
        return {
          ...json,
          playerId: params.mediaPlayerId
        };
      };
    } else if (funcName.startsWith('RtcEngine_')) {
      switch (funcName) {
        case 'RtcEngine_initialize':
          AgoraRtcNg.newIrisApiEngine();
          break;
        case 'RtcEngine_release':
          AgoraRtcNg.callApi({
            funcName,
            params: JSON.stringify(params),
            buffers
          });
          AgoraRtcNg.destroyIrisApiEngine();
          return;
        case 'RtcEngine_sendMetaData':
          // metadata.buffer
          buffers.push(_base64Js.default.fromByteArray(params.metadata.buffer));
          break;
        case 'RtcEngine_sendStreamMessage':
        case 'RtcEngine_sendStreamMessageEx':
          // data
          buffers.push(_base64Js.default.fromByteArray(params.data));
          break;
        case 'RtcEngine_destroyMediaPlayer':
          // @ts-ignore
          params.mediaPlayerId = params.media_player.getMediaPlayerId();
          params.toJSON = function () {
            return {
              playerId: params.mediaPlayerId
            };
          };
          break;
      }
    }
    let ret = AgoraRtcNg.callApi({
      funcName,
      params: JSON.stringify(params),
      buffers
    });
    if (ret !== undefined && ret !== null && ret !== '') {
      ret = JSON.parse(ret);
      if (isDebuggable()) {
        if (typeof ret.result === 'number' && ret.result < 0) {
          console.error('callApi', funcName, JSON.stringify(params), ret);
        } else {
          console.debug('callApi', funcName, JSON.stringify(params), ret);
        }
      }
      return ret;
    }
  } catch (e) {
    if (isDebuggable()) {
      console.error('callApi', funcName, JSON.stringify(params), e);
    } else {
      console.warn('callApi', funcName, JSON.stringify(params), e);
    }
  }
  return {};
}

/**
 * @internal
 */
function emitEvent(eventType) {
  for (var _len = arguments.length, params = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
    params[_key - 1] = arguments[_key];
  }
  DeviceEventEmitter.emit(eventType, ...params);
}
