// react-native-agora/lib/commonjs/internal/MediaPlayerInternal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MediaPlayerInternal = void 0;
var _tsInterfaceChecker = require("ts-interface-checker");
var _AgoraBase = require("../AgoraBase");
var _AgoraMediaBaseImpl = require("../impl/AgoraMediaBaseImpl");
var _IAgoraMediaPlayerImpl = require("../impl/IAgoraMediaPlayerImpl");
var _IAgoraMediaPlayerSourceImpl = require("../impl/IAgoraMediaPlayerSourceImpl");
var _AgoraMediaBaseTi = _interopRequireDefault(require("../ti/AgoraMediaBase-ti"));
var _IAgoraMediaPlayerTi = _interopRequireDefault(require("../ti/IAgoraMediaPlayer-ti"));
var _IAgoraMediaPlayerSourceTi = _interopRequireDefault(require("../ti/IAgoraMediaPlayerSource-ti"));
var _IrisApiEngine = require("./IrisApiEngine");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
const checkers = (0, _tsInterfaceChecker.createCheckers)(_AgoraMediaBaseTi.default, _IAgoraMediaPlayerTi.default, _IAgoraMediaPlayerSourceTi.default);
class MediaPlayerInternal extends _IAgoraMediaPlayerImpl.IMediaPlayerImpl {
  constructor(mediaPlayerId) {
    super();
    _defineProperty(this, "_mediaPlayerId", void 0);
    _defineProperty(this, "_events", new Map());
    this._mediaPlayerId = mediaPlayerId;
  }
  release() {
    MediaPlayerInternal._source_observers.delete(this._mediaPlayerId);
    MediaPlayerInternal._audio_frame_observers.delete(this._mediaPlayerId);
    MediaPlayerInternal._video_frame_observers.delete(this._mediaPlayerId);
    MediaPlayerInternal._audio_spectrum_observers.delete(this._mediaPlayerId);
    this.removeAllListeners();
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IMediaPlaye, _checkers$IMediaPlaye2, _checkers$IMediaPlaye3, _checkers$IAudioSpect;
    if ((_checkers$IMediaPlaye = checkers.IMediaPlayerSourceObserver) !== null && _checkers$IMediaPlaye !== void 0 && _checkers$IMediaPlaye.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$;
      if (((_MediaPlayerInternal$ = MediaPlayerInternal._source_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$ === void 0 ? void 0 : _MediaPlayerInternal$.length) === 0) {
        this.registerPlayerSourceObserver({});
      }
    }
    if ((_checkers$IMediaPlaye2 = checkers.IMediaPlayerAudioFrameObserver) !== null && _checkers$IMediaPlaye2 !== void 0 && _checkers$IMediaPlaye2.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$2;
      if (((_MediaPlayerInternal$2 = MediaPlayerInternal._audio_frame_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$2 === void 0 ? void 0 : _MediaPlayerInternal$2.length) === 0) {
        this.registerAudioFrameObserver({});
      }
    }
    if ((_checkers$IMediaPlaye3 = checkers.IMediaPlayerVideoFrameObserver) !== null && _checkers$IMediaPlaye3 !== void 0 && _checkers$IMediaPlaye3.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$3;
      if (((_MediaPlayerInternal$3 = MediaPlayerInternal._video_frame_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$3 === void 0 ? void 0 : _MediaPlayerInternal$3.length) === 0) {
        this.registerVideoFrameObserver({});
      }
    }
    if ((_checkers$IAudioSpect = checkers.IAudioSpectrumObserver) !== null && _checkers$IAudioSpect !== void 0 && _checkers$IAudioSpect.strictTest({
      [eventType]: undefined
    })) {
      var _MediaPlayerInternal$4;
      if (((_MediaPlayerInternal$4 = MediaPlayerInternal._audio_spectrum_observers.get(this._mediaPlayerId)) === null || _MediaPlayerInternal$4 === void 0 ? void 0 : _MediaPlayerInternal$4.length) === 0) {
        console.error('Please call `registerMediaPlayerAudioSpectrumObserver` before you want to receive event by `addListener`');
        return false;
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    var _this = this;
    this._addListenerPreCheck(eventType);
    const callback = function () {
      for (var _len = arguments.length, data = new Array(_len), _key = 0; _key < _len; _key++) {
        data[_key] = arguments[_key];
      }
      if (data[0] !== _IrisApiEngine.EVENT_TYPE.IMediaPlayer) {
        return;
      }
      if (data[1].playerId === _this._mediaPlayerId) {
        (0, _IAgoraMediaPlayerSourceImpl.processIMediaPlayerSourceObserver)({
          [eventType]: listener
        }, eventType, data[1]);
        (0, _IAgoraMediaPlayerImpl.processIMediaPlayerAudioFrameObserver)({
          [eventType]: listener
        }, eventType, data[1]);
        (0, _IAgoraMediaPlayerImpl.processIMediaPlayerVideoFrameObserver)({
          [eventType]: listener
        }, eventType, data[1]);
        (0, _AgoraMediaBaseImpl.processIAudioSpectrumObserver)({
          [eventType]: listener
        }, eventType, data[1]);
      }
    };
    const subscription = _IrisApiEngine.DeviceEventEmitter.addListener(eventType, callback);
    this._events.set(listener, {
      eventType,
      subscription
    });
    return subscription;
  }
  removeListener(eventType, listener) {
    if (!this._events.has(listener)) return;
    _IrisApiEngine.DeviceEventEmitter.removeSubscription(this._events.get(listener).subscription);
    this._events.delete(listener);
  }
  removeAllListeners(eventType) {
    if (eventType === undefined) {
      this._events.forEach(value => {
        _IrisApiEngine.DeviceEventEmitter.removeAllListeners(value.eventType);
      });
      this._events.clear();
    } else {
      _IrisApiEngine.DeviceEventEmitter.removeAllListeners(eventType);
      this._events.forEach((value, key) => {
        if (value.eventType === eventType) {
          this._events.delete(key);
        }
      });
    }
  }
  getMediaPlayerId() {
    return this._mediaPlayerId;
  }
  registerPlayerSourceObserver(observer) {
    let observers = MediaPlayerInternal._source_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._source_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerPlayerSourceObserver(observer);
  }
  unregisterPlayerSourceObserver(observer) {
    let observers = MediaPlayerInternal._source_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -_AgoraBase.ErrorCodeType.ErrFailed;
    MediaPlayerInternal._source_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterPlayerSourceObserver(observer);
  }
  registerAudioFrameObserver(observer) {
    let observers = MediaPlayerInternal._audio_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._audio_frame_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerAudioFrameObserver(observer);
  }
  unregisterAudioFrameObserver(observer) {
    let observers = MediaPlayerInternal._audio_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -_AgoraBase.ErrorCodeType.ErrFailed;
    MediaPlayerInternal._audio_frame_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterAudioFrameObserver(observer);
  }
  registerVideoFrameObserver(observer) {
    let observers = MediaPlayerInternal._video_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._video_frame_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerVideoFrameObserver(observer);
  }
  unregisterVideoFrameObserver(observer) {
    let observers = MediaPlayerInternal._video_frame_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -_AgoraBase.ErrorCodeType.ErrFailed;
    MediaPlayerInternal._video_frame_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterVideoFrameObserver(observer);
  }
  registerMediaPlayerAudioSpectrumObserver(observer, intervalInMS) {
    let observers = MediaPlayerInternal._audio_spectrum_observers.get(this._mediaPlayerId);
    if (observers === undefined) {
      observers = [];
      MediaPlayerInternal._audio_spectrum_observers.set(this._mediaPlayerId, observers);
    }
    if (!observers.find(value => value === observer)) {
      observers.push(observer);
    }
    return super.registerMediaPlayerAudioSpectrumObserver(observer, intervalInMS);
  }
  unregisterMediaPlayerAudioSpectrumObserver(observer) {
    let observers = MediaPlayerInternal._audio_spectrum_observers.get(this._mediaPlayerId);
    if (observers === undefined) return -_AgoraBase.ErrorCodeType.ErrFailed;
    MediaPlayerInternal._audio_spectrum_observers.set(this._mediaPlayerId, observers.filter(value => value !== observer));
    return super.unregisterMediaPlayerAudioSpectrumObserver(observer);
  }
  getApiTypeFromSetPlayerOptionInInt(key, value) {
    return 'MediaPlayer_setPlayerOption';
  }
  getApiTypeFromSetPlayerOptionInString(key, value) {
    return 'MediaPlayer_setPlayerOption2';
  }
}
exports.MediaPlayerInternal = MediaPlayerInternal;
_defineProperty(MediaPlayerInternal, "_source_observers", new Map());
_defineProperty(MediaPlayerInternal, "_audio_frame_observers", new Map());
_defineProperty(MediaPlayerInternal, "_video_frame_observers", new Map());
_defineProperty(MediaPlayerInternal, "_audio_spectrum_observers", new Map());
