// react-native-agora/lib/commonjs/internal/MediaRecorderInternal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MediaRecorderInternal = void 0;
var _tsInterfaceChecker = require("ts-interface-checker");
var _AgoraBase = require("../AgoraBase");
var _AgoraMediaBaseImpl = require("../impl/AgoraMediaBaseImpl");
var _IAgoraMediaRecorderImpl = require("../impl/IAgoraMediaRecorderImpl");
var _AgoraMediaBaseTi = _interopRequireDefault(require("../ti/AgoraMediaBase-ti"));
var _IrisApiEngine = require("./IrisApiEngine");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
const checkers = (0, _tsInterfaceChecker.createCheckers)(_AgoraMediaBaseTi.default);
class MediaRecorderInternal extends _IAgoraMediaRecorderImpl.IMediaRecorderImpl {
  constructor() {
    super(...arguments);
    _defineProperty(this, "_events", new Map());
  }
  setMediaRecorderObserver(connection, callback) {
    const key = (connection.channelId ?? '') + connection.localUid;
    if (MediaRecorderInternal._observers.has(key)) {
      return _AgoraBase.ErrorCodeType.ErrOk;
    }
    MediaRecorderInternal._observers.set(key, callback);
    return super.setMediaRecorderObserver(connection, callback);
  }
  release() {
    MediaRecorderInternal._observers.clear();
    this.removeAllListeners();
    super.release();
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IMediaRecor;
    if ((_checkers$IMediaRecor = checkers.IMediaRecorderObserver) !== null && _checkers$IMediaRecor !== void 0 && _checkers$IMediaRecor.strictTest({
      [eventType]: undefined
    })) {
      if (MediaRecorderInternal._observers.size === 0) {
        console.error('Please call `setMediaRecorderObserver` before you want to receive event by `addListener`');
        return false;
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = function () {
      if ((arguments.length <= 0 ? undefined : arguments[0]) !== _IrisApiEngine.EVENT_TYPE.IMediaRecorder) {
        return;
      }
      (0, _AgoraMediaBaseImpl.processIMediaRecorderObserver)({
        [eventType]: listener
      }, eventType, arguments.length <= 1 ? undefined : arguments[1]);
    };
    const subscription = _IrisApiEngine.DeviceEventEmitter.addListener(eventType, callback);
    this._events.set(listener, {
      eventType,
      subscription
    });
    return subscription;
  }
  removeListener(eventType, listener) {
    if (!this._events.has(listener)) return;
    _IrisApiEngine.DeviceEventEmitter.removeSubscription(this._events.get(listener).subscription);
    this._events.delete(listener);
  }
  removeAllListeners(eventType) {
    if (eventType === undefined) {
      this._events.forEach(value => {
        _IrisApiEngine.DeviceEventEmitter.removeAllListeners(value.eventType);
      });
      this._events.clear();
    } else {
      _IrisApiEngine.DeviceEventEmitter.removeAllListeners(eventType);
      this._events.forEach((value, key) => {
        if (value.eventType === eventType) {
          this._events.delete(key);
        }
      });
    }
  }
}
exports.MediaRecorderInternal = MediaRecorderInternal;
_defineProperty(MediaRecorderInternal, "_observers", new Map());
