// react-native-agora/lib/commonjs/internal/MusicContentCenterInternal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MusicPlayerInternal = exports.MusicContentCenterInternal = exports.MusicCollectionInternal = void 0;
var _tsInterfaceChecker = require("ts-interface-checker");
var _IAgoraMusicContentCenterImpl = require("../impl/IAgoraMusicContentCenterImpl");
var _IAgoraMusicContentCenterTi = _interopRequireDefault(require("../ti/IAgoraMusicContentCenter-ti"));
var _MediaPlayerInternal = require("./MediaPlayerInternal");
var _IrisApiEngine = require("./IrisApiEngine");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
const checkers = (0, _tsInterfaceChecker.createCheckers)(_IAgoraMusicContentCenterTi.default);
class MusicContentCenterInternal extends _IAgoraMusicContentCenterImpl.IMusicContentCenterImpl {
  constructor() {
    super(...arguments);
    _defineProperty(this, "_events", new Map());
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IMusicConte;
    if ((_checkers$IMusicConte = checkers.IMusicContentCenterEventHandler) !== null && _checkers$IMusicConte !== void 0 && _checkers$IMusicConte.strictTest({
      [eventType]: undefined
    })) {
      if (MusicContentCenterInternal._event_handlers.length === 0) {
        this.registerEventHandler({});
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = function () {
      if ((arguments.length <= 0 ? undefined : arguments[0]) !== _IrisApiEngine.EVENT_TYPE.IMusicContentCenter) {
        return;
      }
      (0, _IAgoraMusicContentCenterImpl.processIMusicContentCenterEventHandler)({
        [eventType]: listener
      }, eventType, arguments.length <= 1 ? undefined : arguments[1]);
    };
    const subscription = _IrisApiEngine.DeviceEventEmitter.addListener(eventType, callback);
    this._events.set(listener, {
      eventType,
      subscription
    });
    return subscription;
  }
  removeListener(eventType, listener) {
    if (!this._events.has(listener)) return;
    _IrisApiEngine.DeviceEventEmitter.removeSubscription(this._events.get(listener).subscription);
    this._events.delete(listener);
  }
  removeAllListeners(eventType) {
    if (eventType === undefined) {
      this._events.forEach(value => {
        _IrisApiEngine.DeviceEventEmitter.removeAllListeners(value.eventType);
      });
      this._events.clear();
    } else {
      _IrisApiEngine.DeviceEventEmitter.removeAllListeners(eventType);
      this._events.forEach((value, key) => {
        if (value.eventType === eventType) {
          this._events.delete(key);
        }
      });
    }
  }
  registerEventHandler(eventHandler) {
    if (!MusicContentCenterInternal._event_handlers.find(value => value === eventHandler)) {
      MusicContentCenterInternal._event_handlers.push(eventHandler);
    }
    return super.registerEventHandler(eventHandler);
  }
  unregisterEventHandler() {
    MusicContentCenterInternal._event_handlers = [];
    return super.unregisterEventHandler();
  }
  release() {
    MusicContentCenterInternal._event_handlers = [];
    super.release();
  }
  createMusicPlayer() {
    // @ts-ignore
    const mediaPlayerId = super.createMusicPlayer();
    return new MusicPlayerInternal(mediaPlayerId);
  }
}
exports.MusicContentCenterInternal = MusicContentCenterInternal;
_defineProperty(MusicContentCenterInternal, "_event_handlers", []);
class _MusicPlayerInternal extends _IAgoraMusicContentCenterImpl.IMusicPlayerImpl {
  constructor(mediaPlayerId) {
    super();
    _defineProperty(this, "_mediaPlayerId", void 0);
    this._mediaPlayerId = mediaPlayerId;
  }
  getMediaPlayerId() {
    return this._mediaPlayerId;
  }
  getApiTypeFromOpenWithSongCode(songCode) {
    let startPos = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    return 'MusicPlayer_open';
  }
}
class MusicPlayerInternal extends _MediaPlayerInternal.MediaPlayerInternal {
  constructor(mediaPlayerId) {
    super(mediaPlayerId);
    // @ts-ignore
    _defineProperty(this, "_musicPlayer", void 0);
    this._musicPlayer = new _MusicPlayerInternal(mediaPlayerId);
  }
  openWithSongCode(songCode, startPos) {
    return this._musicPlayer.openWithSongCode(songCode, startPos);
  }
}
exports.MusicPlayerInternal = MusicPlayerInternal;
class MusicCollectionInternal extends _IAgoraMusicContentCenterImpl.MusicCollectionImpl {
  constructor(musicCollection) {
    super();
    _defineProperty(this, "_musicCollection", void 0);
    this._musicCollection = musicCollection;
  }
  getCount() {
    return this._musicCollection.count;
  }
  getMusic(index) {
    return this._musicCollection.music[index] ?? {};
  }
  getPage() {
    return this._musicCollection.page;
  }
  getPageSize() {
    return this._musicCollection.pageSize;
  }
  getTotal() {
    return this._musicCollection.total;
  }
}
exports.MusicCollectionInternal = MusicCollectionInternal;
